static/src/javascripts/projects/commercial/modules/dfp/Advert.js

```diff
@@ -0,0 +1,134 @@
+// @flow
+import detect from 'lib/detect';
+import { getCurrentTime } from 'lib/user-timing';
+import { defineSlot } from 'commercial/modules/dfp/define-slot';
+import performanceLogging from 'commercial/modules/dfp/performance-logging';
+import breakpointNameToAttribute
+    from 'commercial/modules/dfp/breakpoint-name-to-attribute';
+
+/** A breakpoint can have various sizes assigned to it. You can assign either on
+ * set of sizes or multiple.
+ *
+ * One size       - `data-mobile="300,50"`
+ * Multiple sizes - `data-mobile="300,50|320,50"`
+ */
+const createSizeMapping = (attr: string) =>
+    attr
+        .split('|')
+        .map(
+            size => (size === 'fluid' ? 'fluid' : size.split(',').map(Number))
+        );
+
+const getAdBreakpointSizes = (advertNode: Element): {} =>
+    detect.breakpoints.reduce((sizes, breakpoint) => {
+        const data = advertNode.getAttribute(`data-${breakpointNameToAttribute(breakpoint.name)}`);
+        if (data) {
+            sizes[breakpoint.name] = createSizeMapping(data);
+        }
+        return sizes;
+    }, {});
+
+type Resolver = (x: boolean) => void;
+
+class Advert {
+    startLoading: any;
+    stopLoading: any;
+    startRendering: any;
+    stopRendering: any;
+    id: string;
+    node: Element;
+    sizes: {};
+    size: ?any;
```

Comment:
I think you can be more specific here and define an AdvertSize type:
```
type AvertSize = 'fluid' | Array<number>
```
and then `sizes` would be a `Map<string, AdvertSize>`